WcaOnRails/app/views/competitions/_index_form.html.erb

```diff
@@ -95,4 +128,35 @@
   });
 
   $('#display-<%= @display %>').button('toggle');
+
+  // From http://stackoverflow.com/questions/901115/how-can-i-get-query-string-values-in-javascript/901144#901144
+  function getParameterByName(name, url) {
+    if (!url) url = window.location.href;
+    name = name.replace(/[\[\]]/g, "\\$&");
+    var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
+        results = regex.exec(url);
+    if (!results) return null;
+    if (!results[2]) return '';
+    return decodeURIComponent(results[2].replace(/\+/g, " "));
+  }
+
+  function display_selectors() {
+    $display_class_from_url = getParameterByName('display');
+    switch($display_class_from_url) {
+      case 'list': $display_class = 'list'; break;
+      case 'map': $display_class = 'map'; break;
+      case 'admin': $display_class = 'admin'; break;
+      default: $display_class = 'list'; break;
+    }
+    $('#competition-query-form').removeClass('admin map').addClass($display_class);
```

Comment:
Why are you removing `admin` and `map`, but not `list`? Doesn't that mean we could end up with `list admin` both as classes?

This is a copy of logic from our competitions controller, isn't it? I think this logic belongs in [`index.js.erb`](https://github.com/jfly/worldcubeassociation.org/blob/master/WcaOnRails/app/views/competitions/index.js.erb), where you *will* have access to the `@display` variable. You'll also need a list of all possible `displays`, but I'm sure that's easy to create.